### PR TITLE
Prevent unnecessary onValueChanged call after TagBox open (T838251)

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

**Review by ksercs — DISMISSED** [review]

File: testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js [hunk 1]
```diff
@@ -886,6 +886,52 @@ QUnit.module("the 'onValueChanged' option", moduleSetup, () => {
...
+                        return res.promise();
+                    }
+                }),
+                paging: false
+            }),
+            displayExpr: "name",
+            valueExpr: "id",
+            value: [2, 3],
+            onValueChanged: valueChangeActionSpy,
+            showSelectionControls: true,
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]:
```suggestion
            showSelectionControls: true
```

File: js/ui/tag_box.js [hunk 2]
```diff
@@ -1160,7 +1160,10 @@ const TagBox = SelectBox.inherit({
         });
 
         this._updateWidgetHeight();
-        this.option("value", value);
+
+        if(!equalByValue(this._list._getKeysByItems(e.addedItems), this.option("value"))) {
```
Comment (San4es) [inline js/ui/tag_box.js]:
```suggestion
        if(!equalByValue(this._list.option("selectedItemKeys"), this.option("value"))) {
```
?

File: testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js [hunk 3]
```diff
@@ -886,6 +886,52 @@ QUnit.module("the 'onValueChanged' option", moduleSetup, () => {
...
+                    },
+                    byKey(key) {
+                        const res = $.Deferred();
+                        setTimeout(() => {
+                            res.resolve(key);
+                        }, TIME_TO_WAIT / 4);
+                        return res.promise();
+                    }
+                }),
+                paging: false
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]:
```suggestion
                paginate: false
```
DataSource has no `paging` options. Did you mean [paginate](https://js.devexpress.com/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#paginate)?

File: testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js [hunk 4]
```diff
@@ -5095,7 +5141,7 @@ QUnit.module("performance", () => {
         $(".dx-list-select-all-checkbox").trigger("dxclick");
 
         // assert
-        assert.equal(keyGetterCounter, 613, "key getter call count");
+        assert.equal(keyGetterCounter, 614, "key getter call count");
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]:
```suggestion
        assert.equal(keyGetterCounter, 613, "key getter call count");
```
If we can use `list.option("selectedItemKeys")` only...

**Review by San4es — CHANGES_REQUESTED** [review]

File: testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js (hunk 3 above)
Comment (alexander-kotov-dx) [reply testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]: I copied it from another example in this file

**Review by ksercs — APPROVED** [review]

**Review by San4es — APPROVED** [review]
LGTM

**Pull request merged**